hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/Source.java

```diff
@@ -55,11 +57,17 @@ protected Source(TypedProperties props, JavaSparkContext sparkContext, SparkSess
 
   protected Source(TypedProperties props, JavaSparkContext sparkContext, SparkSession sparkSession,
       SchemaProvider schemaProvider, SourceType sourceType) {
+    this(props, sparkContext, sparkSession, schemaProvider, sourceType, Option.empty());
+  }
+
+  protected Source(TypedProperties props, JavaSparkContext sparkContext, SparkSession sparkSession,
+                   SchemaProvider schemaProvider, SourceType sourceType, Option<StreamProfileSupplier> streamProfilerSupplier) {
```

Comment:
Maybe it's better to have an interface class `StreamContext` wrapping all `SchemaProvider schemaProvider, SourceType sourceType, Option<StreamProfileSupplier> streamProfilerSupplier` so it's flexible to extend the context without changing the constructor once more?